backends/firrtl/firrtl.cc

```diff
@@ -42,6 +43,18 @@ static const FDirection FD_OUT = 0x2;
 static const FDirection FD_INOUT = 0x3;
 static const int FIRRTL_MAX_DSH_WIDTH_ERROR = 20; // For historic reasons, this is actually one greater than the maximum allowed shift width
 
+std::string getFileinfo(const RTLIL::AttrObject *design_entity)
+{
+	std::string src(design_entity->get_src_attribute());
+	std::string fileinfo_str = src.empty() ? "" : "@[" + src + "]";
+
+	// Remove quotes from src attribute as firrtl automatically escapes and
+	// double-quotes them.
+	fileinfo_str.erase(std::remove(fileinfo_str.begin(), fileinfo_str.end(), '\"'), fileinfo_str.end());
```

Comment:
Where do the quotes in the attribute come from?

Reply:
You are entirely right to bring this up.  In the past I was calling the `ILANG_BACKEND::dump_const()` function to dump values to an output stream. When these values were strings, the function used to add quotes around them. The firrtl compiler, in turn, escaped these quotes and the result was double-quoted strings which were hard to read.

However I'm now calling `design_entity->get_src_attribute()` directly and there is no additional quote being put around it, so we can safely remove the unnecessary call to str.erase() here.